libs/platform/src/lib/components/table/table.component.ts

```diff
@@ -211,60 +382,374 @@ export class TableComponent implements OnChanges, OnInit, AfterViewInit, OnDestr
+
         if (alreadySelected) {
             alreadySelected.checked = false;
         }
 
         if (alreadySelected === row) {
             this._unchecked.push(alreadySelected.value);
-            this._emitChange(index);
+            this._emitSelectionChange(index);
 
             return;
         }
 
         this._rows[index].checked = true;
         this._checked.push(row.value);
-        this._emitChange(index);
+        this._emitSelectionChange(index);
     }
 
-    /** @hidden Select/unselect all rows in 'multiple' mode. */
+    /** @hidden Select/Unselect all rows in 'multiple' mode. */
     selectAll(checked: boolean): void {
         this._reset();
-        this.checkedAll = checked;
+        this._checkedAll = checked;
 
         if (checked) {
             this._checkAll();
-            this._emitChange();
+            this._emitSelectionChange();
 
             return;
         }
 
         this._uncheckAll();
-        this._emitChange();
-    }
-
-    /** @hidden */
-    getCellValue(key: string, row: SelectableRow): any {
-        return key.split('.').reduce((a, b) => a[b], row.value);
+        this._emitSelectionChange();
     }
 
     /** @hidden */
     onKeydown(event: KeyboardEvent): void {
         // TODO: since the table should be able the focusable cells, needs to implement arrow buttons navigation in next phases
 
         event.stopPropagation();
-        const click = new MouseEvent('click');
 
         if (KeyUtil.isKeyCode(event, [SPACE, ENTER])) {
+            const click = new MouseEvent('click');
             event.target.dispatchEvent(click);
             event.preventDefault();
         }
     }
 
+    /** @hidden */
+    sort(field: string, direction: SortDirection): void {
+        this._tableService.sort(field, direction);
+        this._popoverOpen = false;
+    }
+
+    /** @hidden */
+    filter(field: string, value: string): void {
+        this._tableService.filter([
+            {
+                field: field,
+                value: value,
+                strategy: CollectionStringFilterStrategy.CONTAINS
+            } as CollectionFilter
+        ]);
+
+        this._popoverOpen = false;
+    }
+
+    /** @hidden */
+    group(field: string): void {
+        this._tableService.group(field, SortDirection.ASC);
+        this._popoverOpen = false;
+    }
+
+    /** @hidden */
+    freezeTo(columnKey: string): void {
+        this._tableService.freezeTo(columnKey);
+        this.freezeColumnsTo = columnKey;
+        this._setFreezableInfo();
+    }
+
+    /** @hidden */
+    unfreeze(columnKey: string): void {
+        const idx = this._freezableColumns.indexOf(columnKey);
+        const freezeToKey = this._freezableColumns[idx - 1];
+        this._tableService.freezeTo(freezeToKey);
+        this.freezeColumnsTo = freezeToKey;
+        this._setFreezableInfo();
+    }
+
+    /** Search */
+    search(searchInput: SearchInput): void {
+        this._tableService.search(searchInput);
```

Comment:
How this search is connected to DataSource, where are you subscribing to search results? I see other subscriptions but not for the search

Reply:
`this._tableService.search(searchInput)` triggers changes in `this._tableService.tableStateChanges$` it leads to calling `dataSource.fetch(state)` in the `_listenToTableStateChanges` method.